### PR TITLE
Manual update to v2

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
There are still some build issues - I wonder if we need more `include_directories(${CMAKE_CURRENT_BINARY_DIR})`? Do you have any ideas what the issue could be @h-vetinari @erip?

/cc @hmaarrfk I think we need this for the PyTorch 2.0.0 enhancements.

**Comment by Tobias-Fischer** [comment]
For reference, one of the errors (with the other ones looking basically the same):
```
 gmake[2]: *** No rule to make target `include/triton/Dialect/TritonGPU/IR/mlir-tblgen', needed by `include/triton/Dialect/TritonGPU/IR/Dialect.h.inc'.  Stop.
```

**Comment by hmaarrfk** [comment]
Try to force oyttorch 2 dependency? At build time maybe?

**Comment by h-vetinari** [comment]
> For reference, one of the errors (with the other ones looking basically the same):

Try including `mlir` in the build deps.

**Comment by Tobias-Fischer** [comment]
Unfortunately that didn’t make a difference. Any other clues?

**Comment by h-vetinari** [comment]
It's possible we need to provide the folder for mlir-tblgen. There's should be other places that do so in c-f already, so a quick search in the org should provide some insights (I'm on a phone right now, so I'm limited with what I can do)

**Comment by Tobias-Fischer** [comment]
I think mlir-tblgen is just an executable in the mlir package. Probably some of the cmake is messed up adding a relative path to it, rather than just calling the executable. I spent some time on it, but could not figure it out :(

**Comment by h-vetinari** [comment]
Yes, it's an executable, that's why I said we need it in build. It's possible we need it in host too. Like for other things like protobuf, it's often possible to override the path where cmake will look for that executable. If upstream triton hardcodes its expectation of where to look for it, then we'll have to patch it

**Comment by Tobias-Fischer** [comment]
I spent some more time getting this to work. Locally it builds fine :).

I am not sure what's wrong with the `mlir_tablegen` function that is shipped with mlir. The new cmake function does the job. I also unvendored pybind11.

Would be great if you take a look @h-vetinari @erip 

/cc @hmaarrfk

**Comment by Tobias-Fischer** [comment]
It fails on the old gcc 7 as it does not know `#include <filesystem>`. Any tips for easiest workaround?

**Comment by h-vetinari** [comment]
> Any tips for easiest workaround?

Skip cuda 10 (which has a hard cap that GCC can't be higher than 7)

**Comment by Tobias-Fischer** [comment]
Thanks - my thought was way too complicated ..

**Comment by Tobias-Fischer** [comment]
Ok all builds passing now - this is ready for review @conda-forge/triton

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,15 +27,20 @@ requirements:
...
     - pip
-    - llvmdev =11
+    - llvmdev =14
     - libzlib
+    - mlir =14
   run:
     - python
     - filelock
     - pytorch =*=cuda*
+    - libllvm14
```
Comment (hmaarrfk) [inline recipe/meta.yaml:42]: Shouldn't this run export happen automatically

**Comment by hmaarrfk** [comment]
Does building with newer llvm break things?

File: recipe/meta.yaml (hunk 1 above)
Comment (Tobias-Fischer) [reply recipe/meta.yaml:42]: No, if that’s not added the test falls with a missing library.

**Comment by Tobias-Fischer** [comment]
> Does building with newer llvm break things?

Haven’t tried - upstream builds with 14 as far as I know.

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:42]: I'm always confused as to the best practices for using LLVM.

For pyside2 we used the package `llvmdev` and added the pinnings to the config. To my knowledge, this is the preferred way of doing it.

https://github.com/conda-forge/pyside2-feedstock/commit/b0a41f3db265a7654751b9a6d992dd5c43b3f93c

**Comment by Tobias-Fischer** [comment]
LLVM16 does not work, fails with errors such as:
```
  CMake Error at /home/conda/feedstock_root/build_artifacts/triton_1682979982771/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/cmake/llvm/AddLLVM.cmake:551 (add_dependencies):
    The dependency target "MLIRArithmetic" of target "obj.TritonIR" does not
    exist.
```

File: recipe/meta.yaml (hunk 1 above)
Comment (Tobias-Fischer) [reply recipe/meta.yaml]: Fails with `ImportError: libLLVM-14.so: cannot open shared object file: No such file or directory` if not added manually.

**Comment by Tobias-Fischer** [comment]
@hmaarrfk could you please take another look?

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:42]: maybe the right package is `llvm` and not `llvmdev`

**Comment by hmaarrfk** [comment]
Do you think that this is a problem:
```
2023-05-02T02:11:25.8451136Z   CMake Warning at /home/conda/feedstock_root/build_artifacts/triton_1682993182968/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/lib/cmake/llvm/AddLLVM.cmake:1752 (message):
2023-05-02T02:11:25.8459801Z     LLVM_EXTERNAL_LIT set to
2023-05-02T02:11:25.8468552Z     /home/conda/feedstock_root/build_artifacts/triton_1682993182968/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/llvm-lit,
2023-05-02T02:11:25.8476628Z     but the path does not exist.
```

I couldn't get this to go away, even with my re-org in https://github.com/conda-forge/triton-feedstock/pull/4

**Comment by Tobias-Fischer** [comment]
I don't think so - lit is just used for testing as far as I know.

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:42]: I'm not sure how you feel about these changes. they seem to work locally, but I cleaned up the history, so maybe the tests might fail:
https://github.com/conda-forge/triton-feedstock/pull/4/commits/59f1b1b741a19e7b01b1c195a4e1b016f0af88e2

**Comment by hmaarrfk** [comment]
Let me know how you feel about https://github.com/conda-forge/triton-feedstock/pull/4/files

i did some minor cleanups.

I also noticed you are not a maintainer. Would you like to be one? If so, maybe cherry pick my changes and add yourself as a maintainer here?

**Comment by h-vetinari** [comment]
> I'm always confused as to the best practices for using LLVM.

For getting a [run-export](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/meta.yaml#L101-L106), depend on `llvm` (design predates my involvement, but it's clearly marked in the feedstock).

> Do you think that this is a problem:
> 
> ```
> LLVM_EXTERNAL_LIT set to $PREFIX/bin/llvm-lit, but the path does not exist.
> ```

lit is mostly about testing AFAIK, but you could add a host-dep on [`lit`](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/meta.yaml#L151).

**Comment by hmaarrfk** [comment]
> For getting a [run-export](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/meta.yaml#L101-L106), depend on llvm (design predates my involvement, but it's clearly marked in the feedstock).

done in #4 

> lit is mostly about testing AFAIK, but you could add a host-dep on [lit](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/meta.yaml#L151).

I feel like i tried locally, but the executable was `lit` not `llvm-lit`. i ended up giving up.

**Comment by h-vetinari** [comment]
> I feel like i tried locally, but the executable was `lit` not `llvm-lit`. i ended up giving up.

Yes, for us it's `$PREFIX/bin/lit` (through the [entry-point](https://github.com/conda-forge/llvmdev-feedstock/blob/main/recipe/meta.yaml#L158)), and you'd have to modify `LLVM_EXTERNAL_LIT`.

**Comment by Tobias-Fischer** [comment]
> Let me know how you feel about https://github.com/conda-forge/triton-feedstock/pull/4/files
> 
> i did some minor cleanups.
> 
> I also noticed you are not a maintainer. Would you like to be one? If so, maybe cherry pick my changes and add yourself as a maintainer here?

I'm happy with your PR in #4 - thanks for putting in the additional work!

No need to add me as a maintainer - I'm hoping this was a once-off effort to get v2 to work :).

**Comment by Tobias-Fischer** [comment]
See #4 instead

**Pull request closed without merging**